### DISCUSSION TITLE
git push origin main fatal: remote error:  sahneyoghurt@googlemail.com/sfguide-terraform-sample is not a valid repository name Visit https://support.github.com/ for help

### DISCUSSION THREAD

**Discussion by sahneyoghurt** [community/community#81471] [New to GitHub] [answered]
Hey Community,
I have tried the following on my gitbash on windows:

$ mkdir sfguide-terraform-sample && cd sfguide-terraform-sample
$ echo "# sfguide-terraform-sample" >> README.md
$ git init
$ git add README.md
$ git commit -m "first commit"
$ git branch -M main
$ git remote add origin git@github.com:sahneyoghurt@googlemail.com/sfguide-terraform-sample.git
ssh-keygen -t ed25519 -C "my username email"
ssh-add ~/.ssh/id_ed25519
I than added that to by github account
$ git push -u origin main

I did not add a passphrase to the ssh key. 

I get 
 git push origin main fatal: remote error:  sahneyoghurt@googlemail.com/sfguide-terraform-sample is not a valid repository name Visit https://support.github.com/ for help

I suspect since the "@" is not allowed but how can I change this? I tried just using "sahneyoghurt" my username to generate another key and put that to github and changed my origin but than I get: No repo / No access whatever error appranently I can't generate a ssh key or use and url with just my username without that "@".

Seems like I fall over some stupid detail here, what makes me wonder the most is, that I can't find a single topic addressing this issue. I can't be the only damn person on the planet having this? 

Anyone able to explain what is going wrong?

**Comment by ananuness** [comment]
Hii! You're just using a wrong ssh url to your repo, as you can see, the structure of the url is:

```
git@github.com:<username>/<repo_name>.git

# with your user:
git@github.com:sahneyoghurt/my-repo.git
```

I hope it helps and don't hesitate to ask anything!

**Reply by sahneyoghurt** [reply]
Hi Ananuness,
Thank you for your reply. Unfortunately I still could not make it work.

I tried:
git remote set-url origin git@github.com:sahneyoghurt/sfguide-terraform-sample.git

when I check I get following:
$ git remote -v
origin  git@github.com:sahneyoghurt/sfguide-terraform-sample.git (fetch)
origin  git@github.com:sahneyoghurt/sfguide-terraform-sample.git (push)

When I try to push now:

$ git push -u origin main
"ERROR: Repository not found.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists."

Do you have any other idea what could be causing this?

**Reply by ananuness** [reply, marked as answer]
Oh, this is related to your credentials and/or your write access. But before everything, you can check if you already have the repository created on github and get the SSH url to make sure everything goes well.

To resolve this, you can check this [step by step](https://docs.github.com/en/repositories/creating-and-managing-repositories/troubleshooting-cloning-errors#error-repository-not-found) and check if everything is ok. 

If everything is fine, then you can remove the origin and add it again with these commands:

```
$ git remote rm origin
$ git remote add origin git@github.com:your-user/your-repo.git
$ git push -u origin main
```

If it doesn't work again, then you can check your credentials in the *Credentials Manager* (in your control panel on windows) and see if you have multiple credentials related to GitHub, if so, you have to remove and try to push again. (I saw this answer [here](https://stackoverflow.com/a/51181578/21769341).

If you have any issues, don't hesitate to ask!

**Reply by sahneyoghurt** [reply]
Hey Ananuness,
thank you very much. I had not created the repository on my GitHub as I thought, this would be done automatically during the push request.

I used ssh -T git@github.com
to verfiy my credentials were correct. I got this response as confirmation:

Hi sahneyoghurt! You've successfully authenticated, but GitHub does not provide shell access.

Once I verified that I did add the repository to my github account using the web UI. 

Thank you again for your patience and helping me out here despite this being a rather basic question :D

**Reply by ananuness** [reply]
Yaaay! I'm glad it worked and that you don't need to re-add your credentials 😅 everything is fine now! And you're welcome, have a nice day!

**Reply by Amjadali2001** [reply]
i m tired i cant understand these concept plz guide me how i push my project in github next js bz i m ubantu 23 user its much complecated i cant leave ubantu but i m tired  plz guide how generated key o to hero step amjadsaid9925@gmail.com send me here
